### PR TITLE
Node js async await tutorial with asynchronous javascript examples

### LINKED ISSUE #608: [Auto]（Node.js Async Await 教程 - 使用异步 JavaScript 示例）
- 原文网址：[node-js-async-await-tutorial-with-asynchronous-javascript-examples](https://www.freecodecamp.org/news/node-js-async-await-tutorial-with-asynchronous-javascript-examples/)
- MarkDown 文件：https://github.com/freeCodeCamp/news-translation/edit/main/chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md

### PR DESCRIPTION
**Pull request by luojiyin1987**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #608

### CODE REVIEW

**Comment by luojiyin1987** [comment]
#608

File: chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md [hunk 1]
```diff
@@ -268,9 +263,9 @@ translateLetter("wooden truck")
 // This would produce the exact same present: wrapped polished wooden truck
 ```
 
-However, promises are not without problems either. Data in each eye of our chain have a different scope and only have access data passed from the immediate previous step or parent scope.
+然而，promises 也不是没有问题的。我们链条上的每一个眼的数据都有不同的范围，并且只能访问从紧邻的前一个步骤或父级范围传递过来的数据。
```
Comment (miyaliu666) [inline chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md:266]: 我们链条中每个节点的数据都有不同的作用域，只能访问从紧邻的前一个步骤或父作用域传递过来的数据。

File: chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md [hunk 2]
```diff
@@ -1,79 +1,74 @@
...
+尽管它很难学，但如果你想使用JavaScript和Node.js构建网络应用程序和服务器，异步编程是至关重要的，因为JS代码**默认是异步的**。
 
-## Asynchronous Programming Fundamentals
+## 异步编程基础知识
 
-So what exactly is the asynchronous processing model, or the `non-blocking I/O` model (which you've likely heard of if you're a Node.js user)?
+那么，究竟什么是异步处理模型，或 "非阻塞I/O "模型（如果你是Node.js用户，你可能已经听说过）？
 
-Here's a TL;DR description: in an async processing model, when your application engine interacts with external parties (like a file system or network), it doesn't wait until getting a result from those parties. Instead, it continues on to subsequent tasks and only comes back to those previous external parties once it's gotten a signal of a result.
+这里有一个很长的，请不要看的描述：在异步处理模型中，当你的应用程序引擎与外部各方（如文件系统或网络）进行交互时，它不会等到从这些各方得到一个结果，才做下一步。相反，它继续进行后续的任务，只有在得到结果的信号后才回到之前的外部各方进行处理。
```
Comment (miyaliu666) [inline chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md:21]: 这里有一个简单明了的描述

File: chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md [hunk 3]
```diff
@@ -201,26 +196,26 @@ function translateCb(instruction) {
...
-[Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) were created to solve these above mentioned problems with callbacks. Promises make sure that JavaScript users:
+[Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) 是为了解决上述回调的问题而创建的。Promises确保了JavaScript用户:
 
--   Stick to a specific convention with their signature `resolve` and `reject` functions.
--   Chain the callback functions to a well-aligned and top-down flow.
+- 坚持用他们的签名`resolve`和`reject`函数进行特定的约定。
+- 将回调函数链接到一个排列整齐的、自上而下的流程。
 
-Our previous example with Santa's workshop preparing presents can be rewritten with promises like so:
+我们之前关于圣诞老人工作室准备礼物的例子可以用这样的承诺来重写:
```
Comment (miyaliu666) [inline chinese/articles/node-js-async-await-tutorial-with-asynchronous-javascript-examples.md:218]: 承诺 => promises

**Pull request merged**